lib/find-node-directory.js

```diff
@@ -0,0 +1,40 @@
+var path = require('path')
+  , log = require('npmlog')
+
+function findNodeDirectory() {
+  // Have a look to see what is above us, to try and work out where we are
+  npm_parent_directory = path.join(__dirname, '../../../..')
+  log.verbose('node-gyp root', 'npm_parent_directory is ' + path.basename(npm_parent_directory))
+  node_root_dir = ""
+
+  log.verbose('node-gyp root', 'Finding node root directory')
+  if (path.basename(npm_parent_directory) == 'lib') {
+    // We are in a node install directory
```

Comment:
I'm not sure I understand when this would happen.   With a path like `/opt/node/lib/node_modules/npm/node_modules/node-gyp/lib`, you arrive at `/opt/node/lib/node_modules`, not `/opt/node/lib`.